ext/fg/js/popup.js

```diff
@@ -120,8 +124,14 @@ class Popup {
         return false;
     }
 
-    async showContent(elementRect, writingMode, type=null, details=null) {
+    async showContent(elementRect, writingMode, type, details, context) {
```

Comment:
Maybe not something worth addressing here, but there are a bunch of terms used for `{...}` arguments in functions throughout the project: `details`, `context`, `params`, and `options`. We may want to address this at some point in the future since the terms are so general and can feel interchangeable.

The reason why this one stands out is that there are two. I understand why a new one had to be added, because one is sent to the frame and one is used directly. However, looking at it from the perspective of someone using the `Popup` class's public API, it would be more straightforward to use a single `details` object. The fact that some of the options are passed to the frame is an implementation detail. However x2, this would require `Popup` to know which `details` are relevant for the frame, and which aren't.

It's potentially a design flaw with the APIs and how they use eachother, so I would consider it out of scope for this change.

Reply:
I agree that this is is somewhat confusing, especially when one of them contains another but there's still another argument with the same name. I hope this PR will help to simplify it a bit though, since options aren't passed as an object from `Frontend` to `Popup` now and every class gets their own options using the context.